src/displayapp/screens/Calculator.cpp

```diff
@@ -0,0 +1,398 @@
+  lv_label_set_text_fmt(resultLabel, "%" PRId64, result);
+  lv_obj_set_size(resultLabel, 200, 20);
+  lv_obj_set_pos(resultLabel, 10, 5);
+
+  valueLabel = lv_label_create(lv_scr_act(), nullptr);
+  lv_label_set_long_mode(valueLabel, LV_LABEL_LONG_CROP);
+  lv_label_set_align(valueLabel, LV_LABEL_ALIGN_RIGHT);
+  lv_label_set_text_fmt(valueLabel, "%" PRId64, value);
+  lv_obj_set_size(valueLabel, 200, 20);
+  lv_obj_set_pos(valueLabel, 10, 35);
+
+  buttonMatrix = lv_btnmatrix_create(lv_scr_act(), nullptr);
+  buttonMatrix->user_data = this;
+  lv_obj_set_event_cb(buttonMatrix, eventHandler);
+  lv_btnmatrix_set_map(buttonMatrix, const_cast<const char**>(buttonMap));
+  lv_btnmatrix_set_one_check(buttonMatrix, true);
+  lv_obj_set_size(buttonMatrix, 238, 180);
+  lv_obj_set_style_local_bg_color(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_DEFAULT, Colors::bgAlt);
+  lv_obj_set_style_local_pad_inner(buttonMatrix, LV_BTNMATRIX_PART_BG, LV_STATE_DEFAULT, 1);
+  lv_obj_set_style_local_pad_top(buttonMatrix, LV_BTNMATRIX_PART_BG, LV_STATE_DEFAULT, 1);
+  lv_obj_set_style_local_pad_bottom(buttonMatrix, LV_BTNMATRIX_PART_BG, LV_STATE_DEFAULT, 1);
+  lv_obj_set_style_local_pad_left(buttonMatrix, LV_BTNMATRIX_PART_BG, LV_STATE_DEFAULT, 1);
+  lv_obj_set_style_local_pad_right(buttonMatrix, LV_BTNMATRIX_PART_BG, LV_STATE_DEFAULT, 1);
+  lv_obj_align(buttonMatrix, nullptr, LV_ALIGN_IN_BOTTOM_MID, 0, 0);
+
+  lv_obj_set_style_local_bg_opa(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, LV_OPA_COVER);
+  lv_obj_set_style_local_bg_grad_stop(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, 128);
+  lv_obj_set_style_local_bg_main_stop(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, 128);
+}
+
+void Calculator::OnButtonEvent(lv_obj_t* obj, lv_event_t event) {
+  if ((obj == buttonMatrix) && (event == LV_EVENT_PRESSED)) {
+    HandleInput();
+  }
+}
+
+void Calculator::HandleInput() {
+  const char* buttonText = lv_btnmatrix_get_active_btn_text(buttonMatrix);
+
+  if (buttonText == nullptr) {
+    return;
+  }
+
+  if ((equalSignPressed && (*buttonText != '=')) || (error != Error::None)) {
+    ResetInput();
+    UpdateOperation();
+  }
+
+  // we only compare the first char because it is enough
+  switch (*buttonText) {
+    case '0':
+    case '1':
+    case '2':
+    case '3':
+    case '4':
+    case '5':
+    case '6':
+    case '7':
+    case '8':
+    case '9': {
+      // *buttonText is the first char in buttonText
+      // "- '0'" results in the int value of the char
+      auto digit = (*buttonText) - '0';
+      auto sign = (value < 0) ? -1 : 1;
+
+      // if this is true, we already pressed the . button
+      if (offset < FIXED_POINT_OFFSET) {
+        value += sign * offset * digit;
+        offset /= 10;
+      } else if (value <= MAX_VALUE / 10) {
+        value *= 10;
+        value += sign * offset * digit;
+      }
+
+      NRF_LOG_INFO(". offset: %" PRId64, offset);
```

Comment:
Are these logging messages still meant to be here? If you think they're worth keeping and are useful then that's fine by me but otherwise they could be removed to make the code shorter and easier to read

Reply:
I used them for debugging, but we can remove them.